int64 num
int32 fsr1
int32 fsr2
int32[] fsr_raw
